src/Workleap.Extensions.OpenAPI.Analyzers/CompareTypedResultWithAnnotationAnalyzer.cs

```diff
@@ -0,0 +1,294 @@
+            {
+                var subType = typedReturnType.TypeArguments[0];
+                if (subType is INamedTypeSymbol namedSubType)
+                {
+                    typedReturnType = namedSubType;
+                }
+            }
+
+            return typedReturnType;
+        }
+
+        private void ValidateAnnotationWithTypedResult(SymbolAnalysisContext context, AttributeData attribute,
+            Dictionary<int, ITypeSymbol> methodSignatureStatusCodeToTypeMap)
+        {
+            if (attribute.AttributeClass == null || attribute.ConstructorArguments.Length == 0)
+            {
+                return;
+            }
+
+            if (attribute.AttributeClass.Equals(this.ProducesResponseSymbol, SymbolEqualityComparer.Default))
+            {
+                if (attribute.ConstructorArguments.Length == 1)
+                {
+                    if (attribute.ConstructorArguments[0].Value is int statusCodeValue)
+                    {
+                        var type = this._statusCodeToResultsMap[statusCodeValue];
+                        ValidateAnnotationForTypeMismatch(context, methodSignatureStatusCodeToTypeMap, statusCodeValue, type, attribute);
+                    }
+                }
+                else if (attribute.ConstructorArguments[1].Value is int statusCodeValue && attribute.ConstructorArguments[0].Value is ITypeSymbol type)
+                {
+                    ValidateAnnotationForTypeMismatch(context, methodSignatureStatusCodeToTypeMap, statusCodeValue, type, attribute);
+                }
+            }
+
+            else if (attribute.AttributeClass.ConstructedFrom.Equals(this.ProducesResponseOfTSymbol, SymbolEqualityComparer.Default))
+            {
+                if (attribute.ConstructorArguments[0].Value is int statusCodeValue)
+                {
+                    ValidateAnnotationForTypeMismatch(context, methodSignatureStatusCodeToTypeMap, statusCodeValue, attribute.AttributeClass.TypeArguments[0], attribute);
+                }
+            }
+
+            else if (attribute.AttributeClass.ConstructedFrom.Equals(this.SwaggerResponseSymbol, SymbolEqualityComparer.Default))
+            {
+                if (attribute.ConstructorArguments.Length > 2 && attribute.ConstructorArguments[0].Value is int statusCodeValue && attribute.ConstructorArguments[2].Value is ITypeSymbol type)
+                {
+                    ValidateAnnotationForTypeMismatch(context, methodSignatureStatusCodeToTypeMap, statusCodeValue, type, attribute);
+                }
+            }
+        }
+
+        private Dictionary<int, ITypeSymbol> GetMethodReturnStatusCodeToTypeMap(ITypeSymbol methodReturnSignature)
+        {
+            var methodReturnSignatures = this.GetReturnStatusAndTypeFromMethodReturn(methodReturnSignature);
+            var methodSignatureStatusCodeToTypeMap = new Dictionary<int, ITypeSymbol>();
+            foreach (var returnValues in methodReturnSignatures)
```

Comment:
Need to rework since won't support multiple Ok<> type for the same endpoint.

Could convert to Dictionary<int, List<ITypedSYmbok>>